examples/instrumentation/go-synthetic/main.go

```diff
@@ -30,36 +30,44 @@ import (
 
 	"github.com/oklog/run"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
 var (
-	addr             = flag.String("listen-address", ":8080", "The address to listen on for HTTP requests.")
-	cpuBurnOps       = flag.Int("cpu-burn-ops", 0, "Operatins per second burning CPU. (Used to simulate high CPU utilization. Sensible values: 0-100.)")
-	memBallastMBs    = flag.Int("memory-ballast-mbs", 0, "Megabytes of memory ballast to allocate. (Used to simulate high memory utilization.)")
-	maxCount         = flag.Int("max-count", labelCombinations, "Maximum metric instance count for all metric types.")
-	histogramCount   = flag.Int("histogram-count", 2, "Number of unique instances per histogram metric.")
-	gaugeCount       = flag.Int("gauge-count", -1, "Number of unique instances per gauge metric.")
-	counterCount     = flag.Int("counter-count", -1, "Number of unique instances per counter metric.")
-	summaryCount     = flag.Int("summary-count", -1, "Number of unique instances per summary metric.")
+	addr                 = flag.String("listen-address", ":8080", "The address to listen on for HTTP requests.")
+	cpuBurnOps           = flag.Int("cpu-burn-ops", 0, "Operations per second burning CPU. (Used to simulate high CPU utilization. Sensible values: 0-100.)")
+	memBallastMBs        = flag.Int("memory-ballast-mbs", 0, "Megabytes of memory ballast to allocate. (Used to simulate high memory utilization.)")
+	maxCount             = flag.Int("max-count", labelCombinations, "Maximum metric instance count for all metric types.")
+	histogramCount       = flag.Int("histogram-count", 2, "Number of unique instances per histogram metric.")
+	nativeHistogramCount = flag.Int("native-histogram-count", -1, "Number of unique instances per native-histogram metric."+
+		"Note that native histograms are not supported in text format exposition, so traditional protobuf format has to be enabled on your collector. See https://prometheus.io/docs/prometheus/latest/feature_flags/#native-histograms")
+	gaugeCount   = flag.Int("gauge-count", -1, "Number of unique instances per gauge metric.")
+	counterCount = flag.Int("counter-count", -1, "Number of unique instances per counter metric.")
+	summaryCount = flag.Int("summary-count", -1, "Number of unique instances per summary metric.")
+
+	omStateSetCount       = flag.Int("om-stateset-count", -1, "Number of OpenMetrics StateSet metrics (https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#stateset). Requires OpenMetrics format to be negotiated.")
```

Comment:
Bit misleading as those as om* flags are not yet implemented (will be in the next PR).